Inc/stm32f446xx_gpio.h

```diff
@@ -11,26 +11,50 @@
 #include<stdint.h>
 #include"stm32f446xx.h"
 
-typedef struct{
 
-	__vo uint8_t GPIO_PinNumber;							// CAN BE ANYONE OF @GPIO_PIN_NUMBERS
-	__vo uint8_t GPIO_PinMode;								// CAN BE ANYONE OF @GPIO_MODES
-	__vo uint8_t GPIO_PinSpeed;								// CAN BE ANYONE OF @GPIO_SPEED
-	__vo uint8_t GPIO_PinPuPdControl;						// CAN BE ANYONE OF @GPIO_PULL_UP_PULL_DOWN
-	__vo uint8_t GPIO_PinOPType;							// CAN BE ANYONE OF @GPIO_OUTPUT_TYPE
-	__vo uint8_t GPIO_PinAltFunMode;
+class GPIO_PinConfig_t:{
+
+	   private:
+             __vo uint8_t GPIO_PinNumber;							// CAN BE ANYONE OF @GPIO_PIN_NUMBERS
+	         __vo uint8_t GPIO_PinMode;								// CAN BE ANYONE OF @GPIO_MODES
+	         __vo uint8_t GPIO_PinSpeed;								// CAN BE ANYONE OF @GPIO_SPEED
+	         __vo uint8_t GPIO_PinPuPdControl;						// CAN BE ANYONE OF @GPIO_PULL_UP_PULL_DOWN
+	         __vo uint8_t GPIO_PinOPType;							// CAN BE ANYONE OF @GPIO_OUTPUT_TYPE
+	         __vo uint8_t GPIO_PinAltFunMode;
+
+
+
+GPIO_PinConfig_t():  GPIO_PinNumber(0), GPIO_PinMode(0),GPIO_PinSpeed(GPIO_SPEED_HIGH),GPIO_PinPuPdControl(GPIO_NO_PUPD	), GPIO_PinOPType(GPIO_OP_TYPE_PP), GPIO_PinAltFunMode(0){}; 
```

Comment:
keep GPIO_PinSPeed = LOW

Reply:
passing macros here is fine r8 or i have to pass the actual values?